File: backend/src/main/java/com/godeltech/mastery/backend/repository/RoleRepository.java
package com.godeltech.mastery.backend.repository;

import com.godeltech.mastery.backend.domain.entity.Role;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface RoleRepository extends JpaRepository<Role, Long> {

  Optional<Role> findRoleByRoleName(String roleName);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/repository/RoleRepository.kt
package com.godeltech.mastery.backend.repository

import com.godeltech.mastery.backend.domain.entity.Role
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface RoleRepository : JpaRepository<Role, Long> {

    fun findRoleByRoleName(roleName: String): Optional<Role>
}